File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
/**
 * 品牌接口
 * @author tanshuai
 *
 */
public interface BrandService {
	
	/**
	 * 查询所有品牌列表
	 * @return
	 */
	List<TbBrand> findAll();
	/**
	 * 品牌分页查询
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(int pageNum,int pageSize);
	
	/**
	 * 添加品牌
	 * @param brand
	 */
	public void add(TbBrand brand);
	
	/**
	 * 根据id查询品牌信息
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	/**
	 * 更新品牌信息
	 * @param brand
	 */
	public void update(TbBrand brand);
	/**
	 * 删除品牌
	 * @param ids
	 */
	public void delete(Long[] ids);
	
	/**
	 * 按照条件查询
	 * @param brand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult search(TbBrand brand,int pageNum,int pageSize);
	
	/**
     * 查询品牌列表:select2需要返回[{id:"",text:""},{id:"",text:""}]类型的数据
     * @return
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//创建品牌服务
app.service("brandService",function($http){
	//查询所有
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	//分页查询
	this.findPage = function(page,size){
		return $http.get('../brand/findPage.do?page='+page+'&size='+size+'')
	}
	//添加
	this.add = function(entity){
		return $http.post('../brand/add.do',entity);
	}
	//更新
	this.update = function(entity){
		return $http.post('../brand/update.do',entity);
	}
	//根据id查询
	this.findOne = function(id){
		return $http.get('../brand/findOne.do?id='+id);
	}
	//根据选中的id数组删除
	this.dele = function(ids){
		return $http.get('../brand/delete.do?ids='+ids);
	}
	//条件查询
	this.search = function(page,size,searchEntity){
		return $http.post('../brand/search.do?page='+page +'&size='+size, searchEntity);
	}
	
	//获取select2需要格式的品牌列表数据
	this.selectOptionList = function(){
		return $http.get('../brand/selectOptionList.do');
	}
});
		